### ISSUE TITLE
Make Alt-F2 dialog capable to run commands

### ISSUE THREAD

**Issue by feskyde** [solus-project/budgie-desktop#265]
Alt-F2 dialog can search for applications, but if you try to run a command, it opens a random application

**Comment by ikeydoherty** [comment]
Random?

**Comment by feskyde** [comment]
Not exactly, my error, it's open the first application starting with the first letter of the comand (i.e. `sudo` opens SuperTuxKart)

**Comment by ikeydoherty** [comment]
Alright, well the old-old rundialog could do this so I see no reason why not

**Comment by X0rg** [comment]
+1, I need this feature me too.

**Comment by JoshStrobl** [comment]
I agree that Alt +F2 should be an option, but personally I'd rather have Budgie Menu extended to have the option to run the input as a command.

The option would appear as a "Run command %s" option and if you type something like "tmux" it would have the following information instead of just "Sorry, no items found."

(Item) Run command "tmux"
(Horizontal Line Break)
(String) No applications found.

The run command would open a terminal and proceed to execute the command.

**Comment by lanmasterlan** [comment]
It would also be nice if one could use the arrow keys to navigate through the proposed applications in the Alt+F2 dialog. Right now we have to change to the mouse.

**Comment by nazar-pc** [comment]
I'm not Solus OS user and even not Budgie Desktop user. I have custom session because of some features lacking in other WMs I'm using Compiz on Ubuntu 16.10.
But I use this run dialog and it is really awesome except it doesn't support execution of arbitrary commands.

As for @JoshStrobl comment - it is reasonable to implement it in both places.

**Comment by michaeltintiuc** [comment]
Any progress on this, was just about to submit the same thing :)

**Comment by ikeydoherty** [comment]
By virtue of the bug _still being open_  - nope.

**Comment by rilian-la-te** [comment]
Try [vala-panel-runner](https://github.com/rilian-la-te/vala-panel/tree/master/runner).

I write it, because I like budgie layout, but want to run commands.

**Comment by michaeltintiuc** [comment]
@ikeydoherty Thanks, at least it's still somewhat considered. It's one of the very few things I missed when made the switch, quite handy when using your mouse on a rare occasion :)

**Comment by nazar-pc** [comment]
I've personally switched to [Ceebro](https://github.com/KELiON/cerebro), you can try it as well. It can also run commands among other things.

**Comment by rilian-la-te** [comment]
Ceebro is JS-written software, so, I will do not use it. (If they rewrite it to GTK C, i will consider usage).
For now, I will use runner, written by myself. It is C-Based, and minics budgie layout.

**Comment by michaeltintiuc** [comment]
I would rather stay as much as possible on a vanilla Budgie and accept things as they are. I prefer to install applets etc only for the features that make no sense to ask @ikeydoherty to implement, like the haste applet, screenshot applet and others.

**Comment by rilian-la-te** [comment]
So, install my GlobalMenu and StatusNotifier applets:) Afaik, Ikey will not implelemt it:)

**Comment by ikeydoherty** [comment]
Please don't speak on my behalf @rilian-la-te  - I didn't say I wouldn't implement it, and the issue is *still open*.

**Comment by rilian-la-te** [comment]
I am sorry.

**Comment by ikeydoherty** [comment]
S'all good. :)

**Comment by EMPAT94** [comment]
Hello! I'm using Manjaro Budgie 21.0.4 (latest as of this comment date). Would love to have the ability to run shell commands and do quick calculations in search bar. 

I see the last comment is around 4 years ago, but the issue is still open. Dare I hope that the feature may still be implemented in future?

PS: I'm new to budgie. Loving the experience. Spreading word to all my friends. Thanks for all the hard work!

**Comment by wxgeo** [comment]
I'm trying Budgie on Debian.
This environment is really nice, stable (contrary to kde 4 I tried briefly last year) and well designed.

However, having to launch a terminal every time I launch a command is a pain.
I have applications (notably Pycharm) installed without using package manager, and when I try to launch them using alt+F2, it launches K3b instead (Why K3b ? I don't know.)

I thought it was some kind of severe regression which would be quickly fixed, since I used alt+F2 to launch commands for 20 years on so many different environments (Gnome 2, Kde3, Unity, Mate, Cinnamon, Gnome3, Kde 4...)

However, when trying to report this, I discover this bug is present since 2015 without being fixed, so I'm a bit surprised...

**Comment by nazar-pc** [comment]
> I have applications (notably Pycharm) installed without using package manager

Use "Tools > Create Desktop Entry..." and you can launch it like any other app without support for commands.

**Comment by wxgeo** [comment]
I have discovered it in the while, but thanks for your advice !
So, for manually installed applications, creating a `.desktop` file in `~/.local/share/applications` seems to be enough to be able to launch them, good to know.

Yet it was surprising this morning to write `pycharm` and to see K3b being launched. ;)
